src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionDescriptorChangeProvider.cs

```diff
@@ -46,28 +47,52 @@ public class PageActionDescriptorChangeProvider : IActionDescriptorChangeProvide
             rootDirectory = rootDirectory.TrimEnd('/');
 
             var importFileAtPagesRoot = rootDirectory + "/" + templateEngine.Options.ImportsFileName;
-            _additionalFilesToTrack = templateEngine.GetImportItems(importFileAtPagesRoot)
-                .Select(item => item.FilePath)
-                .ToArray();
+            var additionalImportFilePaths = templateEngine.GetImportItems(importFileAtPagesRoot)
+                .Select(item => item.FilePath);
+            var pagesRootSearchPattern = rootDirectory + "/**/*.cshtml";
 
-            _searchPattern = rootDirectory + "/**/*.cshtml";
+            if (razorPagesOptions.Value.EnableAreas)
```

Comment:
You should do this by default and unconditionally. We don't have this feature for MVC and razor should be no different. What is your motivation/concern to add it?

Reply:
Associating a controller with an area is opt-in via `AreaAttribute`, so it's not analogous to start with. In addition, we're adding a breaking behavior change to how the page gets routed to so there needs to be a way to disable the feature.

Reply:
Yeah I wish this were on by default, but it **is** a breaking change. We plan to introduce compatibility modes as well, which means that you can just set an app to `2.1 mode` and get all the new stuff.

Reply:
I chatted with @rynowak. While I agree this is a breaking change (as in a behavior change), it relies on two requirements that I don't think would be too common to happen at the same time:
* Change the pages root to be the content root.
* Have an existing Areas folder on your application with pages inside it.

There are a few options here:
* Do the "breaking change" and not provide any option.
* Do the "breaking change" by default and provide the ability to turn it off.
* Do not do the "breaking change" and provide the ability to turn the behavior on with the new compatibility switches functionality.

IMO I would just go ahead and do 1, any other solution adds complexity to this feature with very limited value (based on the assumption made above).

Reply:
I think we'll end up using the new compatibility modes to have our cake and eat it too. That is, this will be off in 2.0 compat mode, but on in 2.1+ compat mode, with ability to turn it on/off in either mode via the option.

Reply:
Fair enough. I'll keep the flag for now and file a work item to replace it once the compat switch feature lands.